crates/wit-component/src/validation.rs

```diff
@@ -564,7 +564,7 @@ fn validate_imports_top_level(
     for (name, ty) in funcs {
         match resolve.worlds[world].imports.get(&world_key(resolve, name)) {
             Some(WorldItem::Function(func)) => {
-                let ty = types.func_type_at(*ty).unwrap();
+                let ty = types[types.core_type_at(*ty)].unwrap_func();
```

Comment:
How does this code know it has a core type index, rather than a component type index?

Reply:
That's contextually known since the `funcs` map comes from parsing a core wasm module (the input to `wasm-tools component new` which is also where the `types` type information comes from.